packages/functional-tests/tests/oauth/handshake.spec.ts

```diff
@@ -0,0 +1,36 @@
+import { test, expect } from '../../lib/fixtures/standard';
+
+test.describe('OAuth and Fx Desktop handshake', () => {
+  test('user signed into browser and OAuth login', async ({
+    target,
+    page,
+    credentials,
+    pages: { login, relier },
+  }) => {
+    await page.goto(
+      target.contentServerUrl +
+        '?context=fx_desktop_v3&entrypoint=fxa%3Aenter_email&service=sync&action=email'
+    );
+    await login.login(credentials.email, credentials.password);
+
+    await relier.goto();
+    await relier.clickEmailFirst();
+
+    // User can sign in with cached credentials, no password needed.
+    await expect(await login.page.innerText('#prefillEmail')).toMatch(
+      credentials.email
+    );
+    await login.page.isVisible('#use-logged-in', { timeout: 100 });
+    await login.submit();
+    await relier.isLoggedIn();
+
+    await relier.signOut();
+
+    // Attempt to sign back in
+    await relier.goto();
+    await relier.clickEmailFirst();
+    await login.page.isVisible('#use-logged-in', { timeout: 100 });
+    await login.submit();
+    await relier.isLoggedIn();
+  });
+});
```

Comment:
This file has 2 tests originally and few more verification steps. Did you merge all that in this one test?